File: core/src/commonMain/kotlin/com/niceoneconsent/sdk/models/ConsentConfig.kt
package com.niceoneconsent.sdk.models

/**
 * Configuration object for initializing the Consent SDK.
 *
 * @property apiKey API key for authenticating with the consent backend
 * @property baseUrl Base URL for the consent API (no trailing slash)
 * @property language Initial language code (e.g., "en", "fr", "de")
 * @property theme UI theme customization; defaults to [ConsentTheme.Light]
 * @property minimumAge Minimum allowed age for consent; defaults to 16
 * @property timeoutMs Network request timeout in milliseconds; defaults to 15000
 * @property retryCount Number of automatic retries on network failure; defaults to 2
 */
data class ConsentConfig(
    val apiKey: String,
    val baseUrl: String,
    val language: String = "en",
    val theme: ConsentTheme = ConsentTheme.Light,
    val minimumAge: Int = 16,
    val timeoutMs: Long = 15000,
    val retryCount: Int = 2
)

File: ios-sdk/Sources/ConsentSDK/Models/ConsentConfig.swift
import Foundation

/// Configuration object for initializing the Consent SDK.
public struct ConsentConfig {
    /// API key for authenticating with the consent backend
    public let apiKey: String
    /// Base URL for the consent API (no trailing slash)
    public let baseUrl: String
    /// Initial language code (e.g., "en", "fr", "de")
    public let language: String
    /// UI theme customization
    public let theme: ConsentTheme
    /// Minimum allowed age for consent
    public let minimumAge: Int
    /// Network request timeout in seconds
    public let timeoutInterval: TimeInterval
    /// Number of automatic retries on network failure
    public let retryCount: Int

    public init(
        apiKey: String,
        baseUrl: String,
        language: String = "en",
        theme: ConsentTheme = .light,
        minimumAge: Int = 16,
        timeoutInterval: TimeInterval = 15,
        retryCount: Int = 2
    ) {
        self.apiKey = apiKey
        self.baseUrl = baseUrl
        self.language = language
        self.theme = theme
        self.minimumAge = minimumAge
        self.timeoutInterval = timeoutInterval
        self.retryCount = retryCount
    }
}
